internal/tiltfile/tiltfile_test.go

```diff
@@ -3295,6 +3295,88 @@ k8s_yaml('foo.yaml')
 	f.loadErrString("ignores must be a string or a sequence of strings; found a starlark.Int")
 }
 
+func TestDockerbuildOnly(t *testing.T) {
+	f := newFixture(t)
+	defer f.TearDown()
+
+	f.dockerfile("Dockerfile")
+	f.yaml("foo.yaml", deployment("foo", image("gcr.io/foo")))
+	f.file("Tiltfile", `
+docker_build('gcr.io/foo', '.', only="myservice/**")
+k8s_yaml('foo.yaml')
+`)
+
+	f.load()
+	f.assertNextManifest("foo",
+		buildFilters("otherservice/bar"),
+		fileChangeFilters("otherservice/bar"),
+		buildMatches("myservice/bar"),
+		fileChangeMatches("myservice/bar"),
+	)
+}
+
+func TestDockerbuildOnlyAsArray(t *testing.T) {
+	f := newFixture(t)
+	defer f.TearDown()
+
+	f.dockerfile("Dockerfile")
+	f.yaml("foo.yaml", deployment("foo", image("gcr.io/foo")))
+	f.file("Tiltfile", `
+docker_build('gcr.io/foo', '.', only=["common/**", "myservice/**"])
+k8s_yaml('foo.yaml')
+`)
+
+	f.load()
+	f.assertNextManifest("foo",
+		buildFilters("otherservice/bar"),
+		fileChangeFilters("otherservice/bar"),
+		buildMatches("myservice/bar"),
+		fileChangeMatches("myservice/bar"),
+		buildMatches("common/bar"),
+		fileChangeMatches("common/bar"),
+	)
+}
+
+func TestDockerbuildInvalidOnly(t *testing.T) {
+	f := newFixture(t)
+	defer f.TearDown()
+
+	f.dockerfile("foo/Dockerfile")
+	f.yaml("foo.yaml", deployment("foo", image("fooimage")))
+
+	f.file("Tiltfile", `
+k8s_resource_assembly_version(2)
+docker_build('fooimage', 'foo', only=[127])
+k8s_yaml('foo.yaml')
+`)
+
+	f.loadErrString("only must be a string or a sequence of strings; found a starlark.Int")
+}
+
+func TestDockerbuildOnlyAndIgnore(t *testing.T) {
+	f := newFixture(t)
+	defer f.TearDown()
+
+	f.dockerfile("Dockerfile")
+	f.yaml("foo.yaml", deployment("foo", image("gcr.io/foo")))
+	f.file("Tiltfile", `
+docker_build('gcr.io/foo', '.', ignore="**/*.md", only=["common/**", "myservice/**"])
```

Comment:
Are there any other weird interactions I should add tests for here?

Reply:
There are probably a bunch. It would be good to say in the docs that only is built on dockerignore, and so may have issues. 

What if you put the same file in both ignore and only?

What if you have elements that start with ! in only?

What if you have \n in strings?

I'm not sure it's worth trying to catch all of these right now, vs. being clear that if you need to be extremely dlever you should talk to us.

Reply:
Oooh these are great. I'm going to add some tests for them, and make myself a TODO to write up some docs about precedence rules and such.

Reply:
For things that don't block this epic, but we should do, it can be better to leave it as a TODO in code than in your notebook so that if you end up working on something else, your thoughts can help the next person to take it up.